app/models/game.rb

```diff
@@ -8,6 +8,22 @@ class Game < ApplicationRecord
 
   scope :available, -> { where('black_player_id IS NULL OR white_player_id IS NULL') }
 
+  def check?
+    
+    self.pieces.each do |piece|
+      color = piece.color
+      
+      if color == 'white'
+        other_king = self.pieces.find_by(type:'King', color: 'black')
+        return true if piece.valid_move?(other_king.row_coordinate, other_king.column_coordinate)
+      elsif color == 'black'
+        other_king = self.pieces.find_by(type:'King', color: 'white')
+        return true if piece.valid_move?(other_king.row_coordinate, other_king.column_coordinate)
+      end
+    end
+    return false
```

Comment:
this can be `false` without the `return`. When on the last line in a Ruby method, the `return` is optional. It always returns whatever the last line is.